app/services/badges/award_community_wellness.rb

```diff
@@ -0,0 +1,65 @@
+module Badges
+  class AwardCommunityWellness
+    REWARD_STREAK_WEEKS = [1, 2, 4, 8, 16, 32].freeze
+
+    def self.call
+      # These are the users 'eligible' to be awarded the badge
+      results = Comments::CommunityWellnessQuery.call
+
+      results.each do |hash|
+        # Parse the serialized results that come from the query
+        weeks_ago = hash["serialized_weeks_ago"].split(",").map(&:to_i)
+        comment_counts = hash["serialized_comment_counts"].split(",").map(&:to_i)
+
+        # `weeks_ago` can have values like the following:
+        #    - [1,2,10,11,12]
+        #    - [5]
+        #    - [1,2,3,4,5,6,7,8]
+        #    - [1,4,17]
+        # We only care for active streak (starting at 1) so we need to filter
+        # these to check how far back the (continuous) streak goes
+        week_streak = 0
+        weeks_ago.each_with_index do |week, index|
+          # Must have 2 or more non-flagged comments posted on that week
+          next unless comment_counts[index] > 1
+
+          # Must be a consecutive streak
+          next unless week_streak + 1 == week
```

Comment:
Good code comment, I scanned this once and knew it should be there but didn't quite process what it did.